npm_and_yarn/lib/dependabot/npm_and_yarn/registry_helper.rb

```diff
@@ -0,0 +1,188 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "yaml"
+require "dependabot/dependency_file"
+require "sorbet-runtime"
+
+module Dependabot
+  module NpmAndYarn
+    class RegistryHelper
+      extend T::Sig
+
+      # Keys for configurations
+      REGISTRY_KEY = "registry"
+      AUTH_KEY = "authToken"
+
+      # Yarn-specific keys
+      NPM_AUTH_TOKEN_KEY_FOR_YARN = "npmAuthToken"
+      NPM_SCOPE_KEY_FOR_YARN = "npmScopes"
+      NPM_REGISTER_KEY_FOR_YARN = "npmRegistryServer"
+
+      # Environment variable keys
+      COREPACK_NPM_REGISTRY_ENV = "COREPACK_NPM_REGISTRY"
+      COREPACK_NPM_TOKEN_ENV = "COREPACK_NPM_TOKEN"
```

Comment:
I suspect setting `COREPACK_NPM_TOKEN_ENV` is not necessary as the Proxy container will inject the needed credentials. But there may be folks out there still using the Ruby directly and handing it tokens. So fine to leave for now.